lib/cocoapods/target/pod_target.rb

```diff
@@ -419,15 +427,18 @@ def _add_recursive_dependent_targets(set)
     # @return [Array<PodTarget>] the recursive targets that this target has a
     #         test dependency upon.
     #
-    def recursive_test_dependent_targets
-      @recursive_test_dependent_targets ||= _add_recursive_test_dependent_targets(Set.new).to_a
+    def recursive_test_dependent_targets(test_spec)
+      @recursive_test_dependent_targets ||= {}
+      @recursive_test_dependent_targets[test_spec] ||= _add_recursive_test_dependent_targets(test_spec, Set.new).to_a
     end
 
-    def _add_recursive_test_dependent_targets(set)
-      test_dependent_targets_by_spec_name.each_value do |dependent_targets|
-        dependent_targets.each do |target|
-          target._add_recursive_dependent_targets(set) if set.add?(target)
-        end
+    def _add_recursive_test_dependent_targets(test_spec, set)
```

Comment:
is this private? can we move it under `private` if not?

Reply:
its `protected`